integration-test/tests/graph_editor.rs

```diff
@@ -29,3 +22,93 @@ async fn create_new_project_and_add_nodes() {
         graph_editor.model.nodes.get_cloned_ref(&added_node_id).expect("Added node is not added");
     assert_eq!(added_node.view.expression.value().to_string(), "");
 }
+
+#[wasm_bindgen_test]
+async fn debug_mode() {
+    let test = IntegrationTestOnNewProject::setup().await;
+    let project = test.project_view();
+    let graph_editor = test.graph_editor();
+
+    assert!(!graph_editor.debug_mode.value());
+
+    // Turning On
+    let expect_mode = project.debug_mode.next_event();
+    let expect_popup_message = project.debug_mode_popup().label().show.next_event();
+    project.enable_debug_mode();
+    assert!(expect_mode.expect());
+    let message = expect_popup_message.expect();
+    assert!(
+        message.contains("Debug Mode enabled"),
+        "Message \"{}\" does not mention enabling Debug mode",
+        message
+    );
+    assert!(
+        message.contains(enso_gui::view::debug_mode_popup::DEBUG_MODE_SHORTCUT),
+        "Message \"{}\" does not inform about shortcut to turn mode off",
+        message
+    );
+    assert!(graph_editor.debug_mode.value());
+
+    // Turning Off
+    let expect_mode = project.debug_mode.next_event();
+    let expect_popup_message = project.debug_mode_popup().label().show.next_event();
+    project.disable_debug_mode();
+    assert!(!expect_mode.expect());
+    let message = expect_popup_message.expect();
+    assert!(
+        message.contains("Debug Mode disabled"),
+        "Message \"{}\" does not mention disabling of debug mode",
+        message
+    );
+    assert!(!graph_editor.debug_mode.value());
+}
+
+#[wasm_bindgen_test]
+async fn zooming() {
+    let test = IntegrationTestOnNewProject::setup().await;
+    let project = test.project_view();
+    let graph_editor = test.graph_editor();
+    let camera = test.ide.ensogl_app.display.scene().layers.main.camera();
+    let navigator = &graph_editor.model.navigator;
+
+    let zoom_on_center = |amount: f32| ZoomEvent { focus: Vector2(0.0, 0.0), amount };
+    let acceptable_range = 0.999..1.001;
+    let zoom_duration_ms = Duration::from_millis(1000);
+
+    // Without debug mode
+    navigator.emit_zoom_event(zoom_on_center(-1.0));
+    sleep(zoom_duration_ms).await;
+    DEBUG!(camera.zoom());
+    assert!(
+        acceptable_range.contains(&camera.zoom()),
```

Comment:
This could be done nicer using [approx](https://docs.rs/approx/latest/approx/index.html), where this would end up being 
`abs_diff_eq!(camera.zoom(), 1.0, epsilon = 0.001);` and thus avoiding the use of the range as an intermediary to indicate the uncertainty.